### ISSUE TITLE
gleam.io

### ISSUE THREAD

**Issue by ArmynC** [lightswitch05/hosts#147]
In issue #90 it was stated that gleam.io was working (directly, no 3rd party). Now it isn't working at all.

Example: [https://gleam.io/iG5U9/population-zero-free-game-key-giveaway]

**Comment by lightswitch05** [comment]
These are advertisements. If you want to look at them, then you’ll need to whitelist it.

**Comment by ArmynC** [comment]
But why it got blacklisted? These aren't intrusive. If you enter the page, you do the tasks or exit.

What type of advertisments? For example they request you to enter a comment on Steam as an example. There aren't ads on page, beside the Giveaway's owner page to reedem.

**Comment by lightswitch05** [comment]
`https://js.gleam.io/oi-w4yN3ez1.js` is loaded by `https://www.tclusa.com/`.

I'll move the root `gleam.io` to the aggressive list. But the the blocked subdomains will stay in the ads and tracking list.

**Comment by ArmynC** [comment]
@lightswitch05 The normal `ads-and-tracking.txt` as I saw, still has `gleam.io`. Shouldn't it be deleted, as the extended one?

**Comment by lightswitch05** [comment]
Nope. That is the list source and is not intended for use. Please only use the extended lists.

**Comment by lightswitch05** [comment]
See #64 for more details

**Comment by ArmynC** [comment]
Yeah, I saw it. I thought it's still usable 

> But if you want a smaller list, the normal base list is available. I hope that answers your questions!

But if you say so, i'll remove it (normal one). In fact, I had the extended one for a long time, only these weeks I tried to replace them with normal ones, because I found them a little too big, sadly. I'll add them back. Thanks!

**Comment by lightswitch05** [comment]
Hosts formatted lists do not support wildcard/pattern matching. The pihole does not support any list formats that have pattern matching/wildcards. I understand not everyone is using pihole, but that is how I use these lists.

Talking about list size, do you use the pihole? When the pihole loads lists, it merges them all into a single list- sorting and removing duplicates. What this means is that the pihole can determine if a host is blocked via a list in a single operation (the computer science term is `O(1)`). In layman’s terms, what that means is that the size of the list has absolutely no impact of performance. It works the same way for whitelists. On the other hand, each and every regex has a measurable performance impact on the pihole. If you have a list of 10 regex filters, the only way for the pihole to determine if `example.com` matches one of those regexes- is to test each one of them. Keep in mind, it works the same way regardless if `example.com` ends up matching the regex or not. In computer science terms, this is a `O(n)` - where `n` is the number of regexes you have. Therefore, the more regex filters you have, the slower your pihole gets- regardless if the regexes are being effective or not. Have a useless entry in a hosts list? It has no impact at all on your system. Have a useless regex that blocks nothing? Well that’s bad news, cause it’s impacting your pihole performance. As others have already found, you can have millions of hosts blocked on a pihole with no impact on the runtime performance of the pihole (admin tasks like updating lists will be slower). Have thousands of regex? Your pihole will be unusable.

**Comment by ArmynC** [comment]
Nope, right now I use a set of hots & filters, which I integrate them in uBlock Origin ([see here](https://github.com/ArmynC/ArminC-uBlock-Settings)). That's why for me it's a bigger list compared to the easier syntax variant. On the other hand, I saw the piHole but never tried to get into it, I see what you mean. A PiHole & Raspberry Pi combination seems interesting. Thanks for the explanation.

**Comment by lightswitch05** [comment]
I have a plan to redesign my hosts tools to incorporate a database with an API server and cron jobs to accomplish some of my manual tasks. If I ever get around to it, it would be possible to generate different list formats without much difficulty. Having the time and motivation is a different problem.
